### PR TITLE
lakectl local: allow skip gitignore update

### LINKED ISSUE #6307: lakectl local init - add optinal flag for .gitignore
Add flag which will skip the update of .gitignore file in a git repository in case the user wants to manage it by themselves

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #6307

### CODE REVIEW

File: cmd/lakectl/cmd/local.go [hunk 1]
```diff
@@ -20,17 +20,22 @@ const (
 	localDefaultSyncPresign     = true
 	localDefaultMinArgs         = 0
 	localDefaultMaxArgs         = 1
+
+	localPresignFlagName     = "presign"
+	localParallelismFlagName = "parallelism"
+	localGitIgnoreFlagName   = "no-gitignore"
```
Comment (arielshaqed) [inline cmd/lakectl/cmd/local.go:26]: Not a fan of negative names for flags.  I would ideally like `gitignore` or `process-gitignore` or any appropriate `VERB-gitignore`, default `true`.

Here this is particularly confusing, as the flagname is the reverse of the meaning of the flag.

**Review by arielshaqed — COMMENTED** [review]
Late to this party, sorry, so not qualified to select approve/request changes.

I have some difficulty understanding this feature.  Why is `.gitignore` special?  I would expect to skip **many** other paths, some of which are considerably more dangerous.  For instance, does package `git` help us avoid pathnames with `..`?  Does it handle pathnames with NULs (`'\0'`) in them?  What about .gitattributes, and .gitmodules?

**Review by arielshaqed — APPROVED** [review]
Approved, thanks.  BUT still worried that gitish files _other_ than `.gitignore` can be silently changed.

File: cmd/lakectl/cmd/local_init.go [hunk 2]
```diff
@@ -54,9 +56,9 @@ var localInitCmd = &cobra.Command{
 	Args:  cobra.RangeArgs(localInitMinArgs, localInitMaxArgs),
 	Run: func(cmd *cobra.Command, args []string) {
 		remote, localPath := getLocalArgs(args, true, false)
-		force := Must(cmd.Flags().GetBool("force"))
-
-		_, err := localInit(cmd.Context(), localPath, remote, force)
+		force := Must(cmd.Flags().GetBool(localForceFlagName))
+		updateIgnore := Must(cmd.Flags().GetBool(localGitIgnoreFlagName))
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/local_init.go:60]: Not sure what you mean. Do we need a default or do we need the flag?

[the remark this answers is not in the document]
Comment (arielshaqed) [reply cmd/lakectl/cmd/local_init.go:60]: I think that the default value for a bool is "false".  AFAIK Cobra lets you define flags with a default value.  So I am asking whether this is the desired default - I think we reversed the sense of this flag so IIRC we toggled the default behaviour?

[the remark this answers is not in the document]
Comment (N-o-Z) [reply cmd/lakectl/cmd/local_init.go:60]: We're explicitly setting the default.
See: `withGitIgnoreFlag`

**Pull request merged**